### PR TITLE
add prometheus ironic exporter

### PR DESCRIPTION
**Pull request by msherman64**
This PR adds the [prometheus ironic exporter](https://opendev.org/openstack/ironic-prometheus-exporter) to export metrics about all nodes known to ironic.

This depends on the following PRs:
- [x] https://github.com/ChameleonCloud/kolla/pull/8
- [x] https://github.com/ChameleonCloud/kolla-containers/pull/32
- [x] https://github.com/ChameleonCloud/kolla-ansible/pull/31

### CODE REVIEW

**Comment by msherman64** [comment]
This lets us correlate metrics with node id/name, instance id/name, user id/name, and project id/name

Example:
image

**Comment by msherman64** [comment]
@diurnalist could I ask for a "big picture" review? It wasn't clear to me which changes should live in e.g. kolla vs kolla-contianers, or kolla-ansible vs chi-in-a-box. I tried to err on the side of:
* "basic functionality goes in kolla, kolla-ansible"
* "customization goes in kolla-containers, chi-in-a-box"

**Review by Mark-Powers — APPROVED** [review]
This looks good to me, and I agree with your reasoning for organizing these changes. I did not see anything of concern really.

File: kolla/defaults.yml [hunk 1]
```diff
@@ -279,6 +283,9 @@ prometheus_server_external_url: "{{ public_protocol }}://{{ prometheus_external_
 prometheus_alertmanager_external_url: "{{ public_protocol }}://{{ prometheus_external_fqdn }}:{{ prometheus_alertmanager_port }}"
 prometheus_bind_address: "{{ lookup('vars', 'ansible_' + network_interface).ipv4.address }}"
 
+enable_prometheus_ironic_exporter: "{{ enable_prometheus | bool and enable_ironic | bool }}"
+# prometheus_ironic_exporter_interval: 60s
```
Comment (diurnalist) [inline kolla/defaults.yml]: Maybe remove this commented line

**Review by diurnalist — APPROVED** [review]
LGTM!

**Comment by diurnalist** [comment]
@msherman64 I think the way you organized it is spot on :) - I left a few little comments but yes, in general that's how I would approach such a patch. Try to put as much in K-A/Kolla as possible, treat it as if it were an upstream contribution, then add Chameleon-specific defaults in chi-in-a-box. Kolla-containers really only needs touching if you need to configure an additional container image to be built (as in this case.) In general I tried to keep overrides in kolla-containers really minimal, and it's probably the hackiest part of the whole thing :)

**Review by codyhammock — APPROVED** [review]
👍🏻

**Pull request merged**